internal/services/recoveryservices/site_recovery_vmware_replicated_vm_resource.go

```diff
@@ -0,0 +1,1204 @@
+
+			processServerId, err := fetchProcessServerIdByName(ctx, fabricClient, fabricId, model.ApplianceName)
+			if err != nil {
+				return fmt.Errorf("fetch process server id: %+v", err)
+			}
+
+			siteID, err := fetchVmwareSiteIdByFabric(ctx, fabricClient, fabricId)
+			if err != nil {
+				return fmt.Errorf("fetch VMWare site id: %+v", err)
+			}
+
+			discoveryMachineId, err := fetchDiscoveryMachineIdBySite(ctx, vmwareMachinesClient, siteID, model.SourceVmName)
+			if err != nil {
+				return fmt.Errorf("fetch discovery machine id %s: %+v", model.SourceVmName, err)
+			}
+
+			runAsAccountId, err := fetchRunAsAccountsIdBySite(ctx, runAsAccountsClient, siteID, model.PhysicalServerCredentialName, model.ApplianceName)
+			if err != nil {
+				return fmt.Errorf("fetch run as account id %s: %+v", model.PhysicalServerCredentialName, err)
+			}
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing site recovery vmware replicated vm %q: %+v", id, err)
+				}
+			}
+
+			if existing.Model != nil {
+				return tf.ImportAsExistsError("azurerm_site_recovery_vmware_replicated_vm", *existing.Model.Id)
+			}
+
+			providerSpecificDetail := replicationprotecteditems.InMageRcmEnableProtectionInput{
+				LicenseType:              pointer.To(replicationprotecteditems.LicenseType(model.LicenseType)),
+				TargetVMName:             &model.TargetVmName,
+				TargetResourceGroupId:    model.TargetResourceGroupId,
+				FabricDiscoveryMachineId: discoveryMachineId,
+				RunAsAccountId:           &runAsAccountId,
+			}
+
+			if model.TargetVmSize != "" {
+				providerSpecificDetail.TargetVMSize = &model.TargetVmSize
+			}
+
+			diskDefaultValueSet := false
+			diskDefaultValue := replicationprotecteditems.InMageRcmDisksDefaultInput{}
+
+			if model.DefaultRecoveryDiskType != "" {
+				diskDefaultValueSet = true
+				diskDefaultValue.DiskType = replicationprotecteditems.DiskAccountType(model.DefaultRecoveryDiskType)
+			}
+
+			if model.DefaultLogStorageAccountId != "" {
+				diskDefaultValueSet = true
+				diskDefaultValue.LogStorageAccountId = model.DefaultLogStorageAccountId
+			}
+
+			if model.DefaultTargetDiskEncryptionSetId != "" {
+				diskDefaultValueSet = true
+				diskDefaultValue.DiskEncryptionSetId = &model.DefaultTargetDiskEncryptionSetId
+			}
+
+			if diskDefaultValueSet {
+				providerSpecificDetail.DisksDefault = &diskDefaultValue
+			}
+
+			if model.TargetAvailabilitySetId != "" {
+				providerSpecificDetail.TargetAvailabilitySetId = pointer.To(model.TargetAvailabilitySetId)
+			}
+
+			if model.TargetZone != "" {
+				providerSpecificDetail.TargetAvailabilityZone = pointer.To(model.TargetZone)
+			}
+
+			if model.TargetBootDiagnosticsStorageAccountId != "" {
+				providerSpecificDetail.TargetBootDiagnosticsStorageAccountId = pointer.To(model.TargetBootDiagnosticsStorageAccountId)
+			}
```

Comment:
Can we group the other `TargetXXX` handlings together to better reason about rather than scatter around?